src/library/scanner/libraryscanner.cpp

```diff
@@ -160,6 +132,34 @@ void LibraryScanner::slotStartScan() {
     kLogger.debug() << "slotStartScan()";
     DEBUG_ASSERT(m_state == STARTING);
 
+    // Clean up the database and fix inconsistencies from previous runs.
+    // See also: https://bugs.launchpad.net/mixxx/+bug/1846945
+    {
```

Comment:
If we wrap it in braces anyway, we could also make it a separate method `cleanupDatabase()`. I think that would make the code more readable and so keeps the diff small if we move it again.